src/routes/CreateSafePage/components/SafeCreationProcess.tsx

```diff
@@ -136,6 +137,12 @@ const createNewSafe = (userAddress: string, onHash: (hash: string) => void): Pro
         // Monitor the latest block to find a potential speed-up tx
         txMonitor({ sender: userAddress, hash: txHash, data: deploymentTx.encodeABI() })
           .then((txReceipt) => {
+            // txMonitor returns the txReceipt from `getTransactionReceipt` which doesn't throw
+            // if it was reverted. We must check the status of the receipt manually.
+            if (didTxRevert(txReceipt)) {
+              reject(new Error('Sped-up tx reverted'))
```

Comment:
You also need to return here because rejecting, unlike throwing, won't exit the function.